src/shared/components/query/QueryStore.ts

```diff
@@ -563,6 +499,181 @@ export class QueryStore
 		default: {}
 	});
 
+
+	private readonly physicalStudiesSet = remoteData<{[studyId:string]:CancerStudy}>({
+		await: ()=>[this.cancerStudies],
+		invoke: async ()=>{
+			return  this.cancerStudies.result.reduce((obj:{[studyId:string]:CancerStudy}, item) =>{
+				obj[item.studyId] = item
+				return obj
+			}, {} as {[studyId:string]:CancerStudy});
+
+		},
+		default: {},
+	})
+
+	readonly userVirtualStudies = remoteData(async ()=>{
+		if (AppConfig.sessionServiceIsEnabled) {
+			return sessionServiceClient.getUserVirtualStudies();
+		} else {
+			return [];
+		}
+	}, []);
+
+	private readonly userVirtualStudiesSet = remoteData<{[studyId:string]:VirtualStudy}>({
+		await: ()=>[this.userVirtualStudies],
+		invoke: async ()=>{
+			return  this.userVirtualStudies.result.reduce((obj:{[studyId:string]:VirtualStudy}, item) =>{
+				obj[item.id] = item
+				return obj
+			}, {});
+
+		},
+		default: {},
+	})
+
+	private readonly sharedVirtualStudiesSet = remoteData<{[studyId:string]:VirtualStudy}>({
+		await: ()=>[this.physicalStudiesSet, this.userVirtualStudiesSet],
+		invoke: async ()=>{
+			let physicalStudiesIdsSet = this.physicalStudiesSet.result
+			let virtualStudiesIdsSet = this.userVirtualStudiesSet.result
+
+			let knownSelectableIds = Object.assign([], Object.keys(physicalStudiesIdsSet), Object.keys(virtualStudiesIdsSet));
+
+			//queried id that are not selectable(this would mostly be shared virtual study)
+			const unknownQueriedIds:string[] = this._defaultSelectedIds.keys().filter(id => !_.includes(knownSelectableIds,id));
+
+			let result:{[studyId:string]:VirtualStudy} = {}
+			
+			await Promise.all(unknownQueriedIds.map(id =>{
+				return new Promise((resolve, reject) => {
+					sessionServiceClient.getVirtualStudy(id).then((virtualStudy)=>{
+						//physical study ids iin virtual study
+						let ids = virtualStudy.data.studies.map(study=>study.id);
+						//unknown/unauthorized studies within virtual study
+						let unKnownPhysicalStudyIds = ids.filter(id => !physicalStudiesIdsSet[id]);
+						if(_.isEmpty(unKnownPhysicalStudyIds)){

```

Comment:
lets not overuse isEmpty.   i believe filter will return empty array if no matches, so just check for length of the array > 0